### PR TITLE
Partial Sector Y decomp

### PR DESCRIPTION
**Pull request by inspectredc**

### CODE REVIEW

File: src/overlays/ovl_i6/fox_sy.c [hunk 1]
```diff
@@ -1,125 +1,118 @@
...
+    Audio_PlaySfx(0x3100000C, actor->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
 }
 
+extern Gfx* D_6014A40;
+
 void func_i6_80197C64(Effect* effect) {
+
     RCP_SetupDL_21();
-    gSPDisplayList(gMasterDisp++, D_6014A40);
+    GDL(&D_6014A40);
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:31]:
```suggestion
gSPDisplayList(gMasterDisp++, D_6014A40);

File: src/overlays/ovl_i6/fox_sy.c [hunk 2]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+    Math_Vec3fFromAngles(&pos, boss->unk_078.x, boss->unk_078.y, 700.0f);
+
+    if ((fabsf(gObjects58[0].obj.pos.x - (boss->obj.pos.x + pos.x)) < 2500.0f) &&
+        (fabsf(gObjects58[0].obj.pos.z - (boss->obj.pos.z + pos.z)) < 5000.0f) &&
+        (fabsf(gObjects58[0].obj.pos.y - (boss->obj.pos.y + pos.y)) < 1800.0f)) {
+        return 1;
+    }
+
+    return 0;
+}
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1069]: bool return type
use true and false for returns :)

File: src/overlays/ovl_i6/fox_sy.c [hunk 3]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+
+extern AnimationHeader D_6003348;
+extern AnimationHeader D_60034C4;
+extern AnimationHeader D_60258A0;
+extern AnimationHeader D_602645C;
+extern AnimationHeader D_60265B4;
+extern AnimationHeader D_602738C;
+extern AnimationHeader D_6029B48;
+extern AnimationHeader D_602B778;
+extern AnimationHeader D_602B8DC;
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1244]: can we put those in at the top of the file?

File: src/overlays/ovl_i6/fox_sy.c [hunk 4]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+                gPlayer[0].unk_228 = 0x20;
+            }
+
+        } else {
+            boss->swork[30] = 0;
+        }
+    }
+}
+
+extern Gfx D_6014BD0;
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1555]: top of the file

File: src/overlays/ovl_i6/fox_sy.c [hunk 5]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+    Vec3f sp10 = { 0.0f, 0.0f, 0.0f };
+
+    if (!(boss->timer_05C & 1)) {
+        if (boss->index == 1) {
+            GPC(100, 100, 255, 255);
+        } else {
+            GPC(255, 255, 100, 255);
+        }
+    } else {
+        GPC(32, 32, 255, 255);
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1568]: use gDPSetPrimColor

File: src/overlays/ovl_i6/fox_sy.c [hunk 6]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+            break;
+        case 18:
+            rot->x += boss->fwork[6];
+            rot->y -= boss->fwork[5];
+            break;
+    }
+    return 0;
+}
+
+extern Gfx D_6013600;
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1597]: top of the file

File: src/overlays/ovl_i6/fox_sy.c [hunk 7]
```diff
@@ -307,108 +257,3067 @@ void func_i6_801983E4(Boss* boss) {
...
+    Boss* boss = (Boss*) data;
+    Vec3f sp38 = { 0.0f, 0.0f, 0.0f };
+
+    if (boss->index != 0) {
+        return func_i6_8019DC4C(limbIndex, dList, pos, rot, boss);
+    }
+    if (boss->timer_05C & 1) {
+        GPC(32, 32, 255, 255);
+    } else {
+        GPC(255, 255, 255, 255);
```
Comment (sonicdcer) [inline src/overlays/ovl_i6/fox_sy.c:1609]: gDPSetPrimColor

**Review by sonicdcer — APPROVED** [review]
amazing work, a few nits tho, i marked some of the shorter macros that shouldn't be used for the final commit, those are only meant for matching a function quickly. I haven't marked them all, but you get the idea :)

**Review by sonicdcer — APPROVED** [review]
Looks good to me :)

File: src/overlays/ovl_i6/fox_sy.c [hunk 8]
```diff
@@ -307,108 +280,3034 @@ void func_i6_801983E4(Boss* boss) {
...
+    if (gHitCount >= 100) {
+        var_s1 = 9;
+    } else {
+        var_s1 = 2;
+    }
+    Rand_SetSeed(1, 0x71AC, 0x263A);
+
+    for (i = 0; i <= (var_s1); i++, actor++) {
+        if (((i != 0) || !(gTeamShields[3] <= 0.0f)) && ((i != 1) || !(gTeamShields[2] <= 0.0f)) &&
+            ((i != 2) || !(gTeamShields[1] <= 0.0f))) {
```
Comment (petrie911) [inline src/overlays/ovl_i6/fox_sy.c:1836]:
```suggestion
        if (((i == 0) && (gTeamShields[3] <= 0.0f)) || ((i = 1) && (gTeamShields[2] <= 0.0f))||
            ((i == 2) || (gTeamShields[1] <= 0.0f))) {
            continue;
            }
```
This gets rid of the negated float compares.

File: src/overlays/ovl_i6/fox_sy.c [hunk 9]
```diff
@@ -307,108 +280,3034 @@ void func_i6_801983E4(Boss* boss) {
...
+            if (boss->swork[34] != 0) {
+                gBossHealthBar = boss->health * 1.7f;
+            } else if (gBosses[0].unk_04A == 0) {
+                gBossHealthBar = (gBosses[1].health + gBosses[2].health) * 1.275f;
+            }
+        }
+        if (D_801615D0.z > 0.0f) {
+            boss->swork[30]++;
+            if (boss->swork[30] > 300) {
+                if (((ABS(D_801615D0.z) + ABS(D_801615D0.x)) < 3500.0f) && (gRadioState == 0)) {
```
Comment (petrie911) [inline src/overlays/ovl_i6/fox_sy.c:1528]: Combinable with &&?

File: src/overlays/ovl_i6/fox_sy.c [hunk 10]
```diff
@@ -307,108 +280,3034 @@ void func_i6_801983E4(Boss* boss) {
...
+            Math_SmoothStepToAngle(&boss->fwork[15], 0.0f, 0.4f, 8.0f, 0.1f);
+            Math_SmoothStepToAngle(&boss->fwork[14], 0.0f, 0.4f, 8.0f, 0.1f);
+        }
+        Math_SmoothStepToF(&boss->fwork[4], 0.0f, 0.2f, 10.0f, 0.0f);
+        Math_SmoothStepToF(&boss->fwork[34], 0.0f, 0.1f, 0.2f, 0.05f);
+        Math_SmoothStepToF(&boss->fwork[41], 0.0f, 0.1f, 0.05f, 0.05f);
+        Math_SmoothStepToF(&boss->fwork[42], 0.0f, 0.1f, 0.05f, 0.05f);
+        Math_SmoothStepToVec3fArray(sp64, boss->vwork, 1, sp1D0, boss->fwork[0], 100.0f, 0.0f);
+        if ((boss->dmgType != 0) && (boss->health > 0)) {
+            boss->dmgType = 0;
```
Comment (petrie911) [inline src/overlays/ovl_i6/fox_sy.c:1474]: dmgType has defines

**Review by petrie911 — COMMENTED** [review]
just some minor things

**Pull request merged**